inference-engine/src/transformations/include/transformations/rt_info/old_api_map_attribute.hpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief Defines old API map attribute
+ * @file old_api_map_attribute.hpp
+ */
+
+#pragma once
+
+#include <assert.h>
+
+#include <functional>
+#include <memory>
+#include <ngraph/attribute_visitor.hpp>
+#include <ngraph/node.hpp>
+#include <ngraph/variant.hpp>
+#include <openvino/core/rtti.hpp>
+#include <set>
+#include <string>
+#include <transformations_visibility.hpp>
+#include <utility>
+
+namespace ov {
+
+class OldApiMap;
+/**
+ * @ingroup ie_runtime_attr_api
+ * @brief OldApiMapAttr class stores the value of OldApiMap class.
+ *
+ *  OldApiMap stores the following information.
+ *  Parameter:
+ *  Order of the transpose which should be applied to Parameter with old API layout to
+ *  obtain Parameter with new API layout.
+ *  Element type of the Parameter in old API.
+ *
+ *  Result:
+ *  Order of the transpose which should be applied to Result with new API layout to
+ *  obtain Result with old API layout.
+ *
+ */
+class TRANSFORMATIONS_API OldApiMapAttr {
+private:
+    std::vector<uint64_t> m_order;
+    ngraph::element::Type m_legacy_type = ngraph::element::Type_t::undefined;
+
+public:
+    friend class OldApiMap;
```

Comment:
Why do we need to introduce friend for `OldApiMapAttr`?

Reply:
All classes for runtime info attributes are derived from VariantImpl template class which needs some type specified for inner value. If attribute has simple value like DisableConstantFolding with bool value, then we don't need a helper class and we can derive from VariantImpl<bool>. But in this case attribute stores several values so we need a helper class to store them.

Reply:
In `visit_attributes()` method of `OldApiMap` class we need to access the private fields of `OldApiMapAttr`. For this reason `OldApiMap` is introduced as friend class.